src/visitor.cpp

```diff
@@ -104,3 +121,65 @@ bool detail::visit(const cpp_entity& e, detail::visitor_callback_t cb, void* fun
     DEBUG_UNREACHABLE(detail::assert_handler{});
     return true;
 }
+
+bool detail::visit(const cpp_entity& e, visitor_filter_t fcb, void* filt_functor,
```

Comment:
Again, I'd prefer it if you implement the non-filtered visit by passing an always true predicate to the filtered one.